mozilla_l10n/activity-stream-convert

```diff
@@ -0,0 +1,110 @@
+    return subprocess.run(['git', 'pull'], cwd=repo, stdout=subprocess.PIPE).stdout.decode('utf-8')
+
+def migrate(locale):
+    repo = os.path.join(args.l10n, locale)
+    out = ''
+    out += 'starting migration for {}\n'.format(locale)
+    if args.method == 'rebase':
+        out += hg_pull(repo)
+    fd, filemap = mkstemp(text=True)
+    try:
+        with os.fdopen(fd, 'w') as fh:
+            fh.write('''\
+    include "l10n/{loc}/strings.properties"
+    rename "l10n/{loc}/strings.properties" "browser/extensions/activity-stream/activity-stream.properties"
+    '''.format(loc=locale))
+        shamap = os.path.join(args.l10n, locale, '.hg', 'shamap')
+        if os.path.isfile(shamap):
+            os.remove(shamap)
+        if args.method == 'rebase':
+            content = subprocess.run(
+                ['hg', 'log', '-r', 'default',
+                 "-Tf9f7148c720362366d60d3e056f08bc76e7c3faa {node}\n"
+                ],
+                cwd=repo, stdout=subprocess.PIPE).stdout.decode('utf-8')
+            with open(shamap, 'w') as fh:
+                fh.write(content)
+        # ignore all the commit messages in each locale, don't add to output
+        subprocess.run(
+            ['hg', 'convert', '--filemap', filemap, args.a_s, repo],
+            stdout=subprocess.PIPE
+        ).stdout.decode('utf-8')
+        if args.method == 'merge':
+            out += hg_pull(repo)
+            heads = json.loads(subprocess.run(
+                ['hg', 'heads', '-Tjson', 'default'],
+                cwd=repo,
+                stdout=subprocess.PIPE
+            ).stdout)
+            if len(heads) == 2:
+                out += subprocess.run(
+                    ['hg', 'up', '-r', heads[0]['node']],
+                    cwd=repo,
+                    stdout=subprocess.PIPE
+                ).stdout.decode('utf-8')
+                out += subprocess.run(
+                    ['hg', 'merge', heads[1]['node']],
+                    cwd=repo,
+                    stdout=subprocess.PIPE
+                ).stdout.decode('utf-8')
+                out += subprocess.run(
+                    ['hg', 'ci', '-m', 'merging localization of activity stream'],
+                    cwd=repo,
+                    stdout=subprocess.PIPE
+                ).stdout.decode('utf-8')
+        if args.push:
+            out += subprocess.run(
+                ['hg', 'push', '-r', 'default'],
+                cwd=repo,
+                stdout=subprocess.PIPE
+            ).stdout.decode('utf-8')
+    finally:
+        os.remove(filemap)
+    out += 'finished migration for {}\n'.format(locale)
+    return out
+
+sys.stdout.write(git_pull(args.a_s))
+locales = [
+    dir
+    for dir in os.listdir(os.path.join(args.a_s, 'l10n'))
+    if dir != 'templates'
+]
+locales.sort()
+handle = []
+skip = []
+for loc in locales:
+    if os.path.isdir(os.path.join(args.l10n, loc)):
+        handle.append(loc)
+    else:
+        skip.append(loc)
+
+handle = handle[:4]
```

Comment:
Why this? I've never used ThreadPoolExecutor, but it seems like you're cutting down the list to 4 locales, not just taking 4 at a time.